### PR TITLE
usnic: allow FI_WAIT_UNSPEC

### PR DESCRIPTION
**Pull request by goodell**
Also, deliberately error on calls to fi_cq_sread when FI_WAIT_NONE has
been specified.  It's not as user friendly in some ways, but I think
taking a stricter interpretation of the libfabric docs is helpful to
users and the project in the long run so that users do not get the
incorrect impression that it is OK to use NONE+sread together.

See discussion in #1629 for more info.

Signed-off-by: Dave Goodell <dgoodell@cisco.com>

@bturrubiates please review

### CODE REVIEW

File: prov/usnic/src/usdf.h [hunk 1]
```diff
@@ -387,6 +387,8 @@ struct usdf_cq {
 		} soft;
 	} c;
 	struct usd_completion cq_comp;
+
+	enum fi_wait_obj cq_wait_obj;
```
Comment (bturrubiates) [inline prov/usnic/src/usdf.h]: so, this is already available through `cq_attr->wait_obj`.

File: prov/usnic/src/usdf_cq.c [hunk 2]
```diff
@@ -244,6 +244,10 @@ usdf_cq_sread_common(struct fid_cq *fcq, void *buf, size_t count, const void *co
 	sleep_time_us = SREAD_INIT_SLEEP_TIME_US;
 
 	cq = cq_ftou(fcq);
+
+	if (cq->cq_wait_obj == FI_WAIT_NONE)
+		return -FI_EINVAL;
```
Comment (bturrubiates) [inline prov/usnic/src/usdf_cq.c]: maybe `FI_EOPNOTSUPP`?

**Comment by goodell** [comment]
updated, good catches

**Comment by bturrubiates** [comment]
:+1:

**Pull request merged**